apps/desktop-wallet/src/api/addressQueries.ts

```diff
@@ -28,9 +29,32 @@ interface AddressBalanceQueryProps {
 // Adding latestTxHash in queryKey ensures that we'll refetch when new txs arrive.
 // Adding networkId in queryKey ensures that switching the network we get different data.
 // TODO: Should we add explorerBackendUrl instead?
-export const addressBalanceQuery = ({ addressHash, networkId, latestTxHash }: AddressBalanceQueryProps) =>
+export const addressAlphBalanceQuery = ({ addressHash, networkId, latestTxHash }: AddressBalanceQueryProps) =>
   queryOptions({
-    queryKey: ['address', 'balance', { addressHash, latestTxHash, networkId }],
+    queryKey: ['address', 'balance', 'ALPH', { addressHash, latestTxHash, networkId }],
     queryFn: () => client.explorer.addresses.getAddressesAddressBalance(addressHash),
     staleTime: Infinity
   })
+
+export const addressTokensBalanceQuery = ({ addressHash, networkId, latestTxHash }: AddressBalanceQueryProps) =>
+  queryOptions({
+    queryKey: ['address', 'balance', 'tokens', { addressHash, latestTxHash, networkId }],
```

Comment:
A few comments about using latestTxHash in the query key:
1. It is a very clean and straightforward way to refetch the query when the last tx hash changes ✅ When the last tx hash changes, a NEW query is being created and the previous won't be used anymore.
2. Setting `staleTime: Infinity` is necessary so that we don't fetch unnecessarily if there is no new tx hash. This minimizes the calls to explorer backend ✅. However, when there will be a new latest tx hash, the previous query will never get garbage collected 🤔.
3. When the tx hash changes, the new query will not prefill its catch with the data from the latest query unless we initialize it with `initialData`. We could keep track of the previous from last TX hash so that we can do that.

I will explore some alternative ways of achieving our goals, as we discussed a few weeks ago.